### PR TITLE
Fixed relative version calculation for non-numeric values

### PR DESCRIPTION
**Pull request by syzygymsu**

### CODE REVIEW

**Comment by mikeSimonson** [comment]
Any test ?

**Comment by mikeSimonson** [comment]
@syzygymsu Thanks, I will look that up next week. Don't hesitate to ping me if I forget about it.

File: lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php [hunk 1]
```diff
@@ -601,15 +601,15 @@ public function getRelativeVersion($version, $delta)
...
+        $versions = array_map('strval', array_keys($this->migrations));
+        array_unshift($versions, '0');
+        $offset = array_search((string)$version, $versions);
         if ($offset === false || !isset($versions[$offset + $delta])) {
             // Unknown version or delta out of bounds.
             return null;
         }
 
-        return (string) $versions[$offset + $delta];
+        return $versions[$offset + $delta];
```
Comment (mikeSimonson) [inline lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php]: I would keep the casting to string there.

Comment (syzygymsu) [inline lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php]: What for? All values are already converted by `strval` and non-string case is covered by tests.
My idea was to fix type discrepancy as soon as possible and do not rely on extra casts.

**Comment by syzygymsu** [comment]
BTW, should we also fix #457 here? It could be covered by this test with just a little change.

File: lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php (hunk 1 above)
Comment (mikeSimonson) [inline lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php]: Sorry I missed it.

**Comment by mikeSimonson** [comment]
If you could add the test for it it would be perfect.

**Comment by syzygymsu** [comment]
Modified test proves that #457 is fixed. I hope we don't need to test that `ksort` realy sorts.

File: tests/Doctrine/DBAL/Migrations/Tests/ConfigurationTest.php [hunk 2]
```diff
@@ -139,7 +139,7 @@ public function testRelativeVersion()
         $config->registerMigrations([
             '0initial' => Version1Test::class,
             1 => Version2Test::class,
-            '2final' => Version3Test::class,
+            'final' => Version3Test::class,
```
Comment (mikeSimonson) [inline tests/Doctrine/DBAL/Migrations/Tests/ConfigurationTest.php]: did the test broke with the number prepending the migration name ?

Comment (syzygymsu) [inline tests/Doctrine/DBAL/Migrations/Tests/ConfigurationTest.php]: Without the number. Yep, until added `SORT_STRING`.

**Pull request merged**